File: presentation/src/main/java/dev/olog/presentation/utils/rx/RxSlidingUpPanel.java
package dev.olog.presentation.utils.rx;

import android.support.annotation.NonNull;

import com.sothree.slidinguppanel.SlidingUpPanelLayout;

import io.reactivex.Observable;

public class RxSlidingUpPanel {

    public static Observable<SlidingPanelState> panelStateEvents(@NonNull SlidingUpPanelLayout slidingUpPanelLayout){
        return new SlidingUpPanelEventsObservable(slidingUpPanelLayout);
    }

}

File: presentation/src/main/java/dev/olog/presentation/utils/rx/RxSlidingUpPanel.kt
package dev.olog.presentation.utils.rx

import com.sothree.slidinguppanel.SlidingUpPanelLayout

import io.reactivex.Observable

object RxSlidingUpPanel {

    fun panelStateEvents(slidingUpPanelLayout: SlidingUpPanelLayout): Observable<SlidingPanelState> {
        return SlidingUpPanelEventsObservable(slidingUpPanelLayout)
    }

}
